x-pack/plugins/serverless_search/public/application/components/connectors/edit_service_type.tsx

```diff
@@ -107,15 +227,40 @@ export const EditServiceType: React.FC<EditServiceTypeProps> = ({ connector, isD
       data-test-subj="serverlessSearchEditConnectorType"
       fullWidth
     >
-      <EuiSuperSelect
-        // We only want to allow people to set the service type once to avoid weird conflicts
-        disabled={Boolean(connector.service_type) || isDisabled}
-        data-test-subj="serverlessSearchEditConnectorTypeChoices"
+      <EuiComboBox<CustomOptionData>
+        aria-label={i18n.translate(
+          'xpack.serverlessSearch.connectors.chooseConnectorSelectable.euiComboBox.accessibleScreenReaderLabelLabel',
+          { defaultMessage: 'Select a data source for your connector to use.' }
+        )}
+        isDisabled={Boolean(connector.service_type) || isDisabled}
         isLoading={isLoading}
-        onChange={(event) => mutate(event)}
-        options={options}
-        valueOfSelected={connector.service_type || undefined}
+        data-test-subj="serverlessSearchEditConnectorTypeChoices"
+        prepend={
+          <EuiIcon
+            type={
+              connector.service_type
+                ? connectorTypes.find((conn) => conn.serviceType === connector.service_type)
+                    ?.iconPath ?? ''
+                : `${assetBasePath}/connectors.svg`
+            }
+            size="l"
+          />
+        }
+        singleSelection={{ asPlainText: true }}
         fullWidth
+        placeholder={i18n.translate(
+          'xpack.serverlessSearch.connectors.chooseConnectorSelectable.placeholder.text',
+          { defaultMessage: 'Choose a data source' }
+        )}
+        options={initialOptions}
+        selectedOptions={initialOptions
+          .filter((option) => option.value?.serviceType === connector.service_type)
+          .slice(0, 1)}
```

Comment:
What are we doing here? this compilation should not be done here. Additionally the `filter().slice()` is confusing me.

If you're trying to find the first item in `initialOptions` that matches `connector.service_type` then `initialOptions.find((option) => option.value?.serviceType === connector.service_type)` would be better. 

depending on how big `initialOptions` is you either should assign that to a const or do it in a `useMemo()`

Avoid doing computational work in JSX.

Reply:
Done using your suggestion and `useMemo()`